src/__tests__/Document.test.ts

```diff
@@ -701,6 +701,60 @@ describe('buildForeignKeyDefinitions', () => {
 		expect(document.buildForeignKeyDefinitions()).toEqual(expected);
 	});
 
+	test('should create foreign key definitions for nested object', () => {
```

Comment:
Did not building foreign key definitions for nested objects/embedded types impact anything in the services? I took a brief look and I don't think it did. I would imagine if it did we'd have gotten some bad data already.

Reply:
There wasn't a problem with the FK and nested objects. There was a problem with embedded schemas, which I don't think have ever been used.